### PR TITLE
Action implementation

### PR DESCRIPTION
**Pull request by BenLanglois**
Don't delete the branch after plz

### CODE REVIEW

File: game/confusion.h [hunk 1]
```diff
@@ -9,4 +9,4 @@ class Confusion: public Mobility {
     Confusion(Hackmon &);
 };
 
-#endif
\ No newline at end of file
+#endif
```
Comment (alexgirard) [inline game/confusion.h:12]: gotta love atom and the auto newline at end of file

File: game/move.cc [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+uniform_int_distribution<unsigned int> Move::random_distribution {1, 100};
+
+Move::Move(const string name, const Scope scope, const unsigned accuracy, const Family family):
+  Action{name, scope}, hackmon{nullptr}, accuracy{accuracy}, family{family} {}
+
+void Move::doActionOverride(Player &player, const std::vector<size_t> &targets) const {
+  // requires: targets must be distinct and attachHackmon must have already been called
+
+  if (!hackmon->mobility->canMove()) {
+    // Print "can't move" or something
```
Comment (alexgirard) [inline game/move.cc]: maybe add `FIXME:` in the comment?

File: game/player.cc [hunk 3]
```diff
@@ -7,22 +7,24 @@
 #include "statInfo.h"
 using namespace std;
 
+unsigned numberBattling; // REMOVE THIS LATER
```
Comment (alexgirard) [inline game/player.cc]: can we remove this before merging? I have it on my game logic branch

File: game/statMove.cc [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  Move{name, scope, accuracy, family}, stat{stat}, strength{strength} {}
+
+void StatMove::doMoveOverride(Hackmon &target) const {
+  if (stat == HP) {
+    // damage is proportional to strength, effectiveness, and the attacker's Attack stat
+    // damage is inversely proportional to the target's Defense stat
+    // damage cannot be less than 1
+
+    int attackStat = hackmon->stats.getStat(ATTACK);
+    int defenseStat = hackmon->stats.getStat(DEFENSE);
```
Comment (alexgirard) [inline game/statMove.cc]: shouldnt it be target in defense stat?

Comment (alexgirard) [reply game/statMove.cc]: better variable names

**Review by alexgirard — CHANGES_REQUESTED** [review]
Few small changes, otherwise it looks good!

File: game/statMove.cc [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    int damage = max(1, strength
+                      * (int) family.effectiveness(target.family)
+                      * max(1, attackerAttackStat)
+                      / max(1, targetDefenseStat)
+    );
+
+    target.stats.setStat(HP, target.stats.getStat(HP) - damage);
+
+  } else {
+
```
Comment (ariellam) [inline game/statMove.cc:39]: ?????????????????

Comment (ariellam) [reply game/statMove.cc:39]: _jgN########Ngg_
      _N##N@@""  ""9NN##Np_
     d###P            N####p
     "^^"              T####
                       d###P
                    _g###@F
                 _gN##@P
               gN###F"
              d###F
             0###F
             0###F
             0###F
             "NN@'

              ___
             q###r
              ""

**Review by ariellam — APPROVED** [review]

**Pull request merged**